src/validation.cpp

```diff
@@ -2225,13 +2224,28 @@ bool CChainState::FlushStateToDisk(
         CoinsCacheSizeState cache_state = GetCoinsCacheSizeState();
         LOCK(m_blockman.cs_LastBlockFile);
         if (fPruneMode && (m_blockman.m_check_for_pruning || nManualPruneHeight > 0) && !fReindex) {
-            // make sure we don't prune above the blockfilterindexes bestblocks
+            // make sure we don't prune above any of the prune locks bestblocks
             // pruning is height-based
             int last_prune = m_chain.Height(); // last height we can prune
+            std::optional<std::string> limiting_lock; // prune lock that actually was the limiting factor, only used for logging
+
             ForEachBlockFilterIndex([&](BlockFilterIndex& index) {
-               last_prune = std::max(1, std::min(last_prune, index.GetSummary().best_block_height));
+                last_prune = std::max(1, std::min(last_prune, index.GetSummary().best_block_height));
             });
 
+            for (const auto& prune_lock : m_blockman.m_prune_locks) {
+                if (prune_lock.second.m_height_first == std::numeric_limits<int>::max()) continue;
+                const int lock_height{prune_lock.second.m_height_first - PRUNE_LOCK_BUFFER};
```

Comment:
In commit "blockstorage: Add prune locks to BlockManager" (d249636878b39aa34222d335cd85f0e2561bd50a)

I think there is a minor off-by-one error here and this should be `lock_height = height_first - PRUNE_LOCK_BUFFER - 1`.

If `height_first` is 100 ("oldest block that must not be pruned"), and `PRUNE_LOCK_BUFFER` is 1 ("number of blocks to keep below the deepest prune lock"), then `last_prune` ("last height we can prune") should be `98` not `99`. Pruning up to 99 when 100 is the first block that can't be pruned would be leaving a buffer of size 0 not size 1.

Reply:
Yeah, makes sense, I also added a comment here to make it clearer to future readers what the additional `- 1` is there for.